addons/mass_mailing/static/src/js/website_mass_mailing.js

```diff
@@ -56,6 +56,68 @@
         },
     });
 
+    website.snippet.animationRegistry.newsletter_popup = website.snippet.Animation.extend({
+        selector: ".o_newsletter_popup",
+        start: function (editable_mode) {
+            var self = this;
+            self.redirect_url = false;
+            var popupcontent = self.$target.find(".o_popup_content_dev").empty();
+
+            openerp.jsonRpc('/website_mass_mailing/get_content', 'call', {
+                newsletter_id: self.$target.data('list-id')
+            }).then(function (data) {
+                if (data.content) {
+                    $('<div></div>').append(data.content).appendTo(popupcontent)
+                }
+                self.$target.find('input.popup_subscribe_email').val(data.email || "")
+                self.redirect_url = data.redirect_url
+                if (!editable_mode && !data.is_subscriber) {
+                    $(document).on('mouseleave', _.bind(self.show_banner, self));
+
+                    self.$target.find('.popup_subscribe_btn').on('click', function (event) {
+                        event.preventDefault();
+                        self.on_click_subscribe();
+                    });
+                } else { $(document).off('mouseleave'); }
+            });
+        },
+        on_click_subscribe: function () {
+            var self = this;
+            var $email = self.$target.find(".popup_subscribe_email:visible");
+
+            if ($email.length && !$email.val().match(/.+@.+/)) {
+                self.$target.addClass('has-error');
+                return false;
+            }
+            self.$target.removeClass('has-error');
+
+            openerp.jsonRpc('/website_mass_mailing/subscribe', 'call', {
+                'list_id': self.$target.data('list-id'),
+                'email': $email.length ? $email.val() : false
+            }).then(function (subscribe) {
+                self.$target.find('#newsletter_popup').modal('hide');
+                $(document).off('mouseleave');
+                if (self.redirect_url) {
+                    if (_.contains(self.redirect_url.split('/'), window.location.host) || self.redirect_url.indexOf('/')== 0) {
+                        window.location.href = self.redirect_url
+                    } else if (self.redirect_url.indexOf('www') == 0) {
```

Comment:
hello @tde-banana-odoo , here check for URL if redirect_url is our internal url then it will open in Same Page and if it is external url then Open In new Tab but If extenal url start With Http:// then there is no problem but start with www. then it will be appened on Our BASE url and  It's Open Url sometinh Like : http://localhost:8069/www.google.com !!! so added Http:// if external url start with www. So is it right behaviour?? Thanks